clients/drcachesim/tools/filter/record_filter.cpp

```diff
@@ -143,20 +144,36 @@ record_filter_tool_create(const std::string &output_dir, uint64_t stop_timestamp
                 new dynamorio::drmemtrace::trim_filter_t(trim_before_timestamp,
                                                          trim_after_timestamp)));
     }
+    if (encoding_filter_enabled) {
+        filter_funcs.emplace_back(
+            std::unique_ptr<dynamorio::drmemtrace::record_filter_t::record_filter_func_t>(
+                new dynamorio::drmemtrace::encoding_filter_t()));
+    }
+
     // TODO i#5675: Add other filters.
 
-    return new dynamorio::drmemtrace::record_filter_t(output_dir, std::move(filter_funcs),
-                                                      stop_timestamp, verbose);
+    /* If we are changing the encoding of trace_entry_t, we will be generating
+     * discrepancies between encoding size and instruction length.  So, we need to tell
+     * reader_t, which here comes in the form of memref_counter_t, to ignore such
+     * discrepancies.  Note that simulators that deal with these filtered traces will
+     * also have to handle the fact that encoding_size != instruction_length.
+     */
+    bool ignore_encoding_size_vs_instr_length_check = encoding_filter_enabled;
```

Comment:
Alternative to remove this extra argument to `record_filter_t` constructor: loop through the `filter_funcs` in the `record_filter_t` constructor and set `ignore_encoding_size_vs_instr_length_check` there if we have a `record_filter_func_t` that is `encoding_filter_t`.

Reply:
See my other comment: this has to be set for every analyzer on the final trace, right?  Just setting it here is not enough.  I think this flag should be removed from here and we need to instead add a filetype and have the reader relax the check on its own based on that, and then have the reader derivative used here somehow set the filetype instead of having a separate mechanism.

Reply:
Yes, now that we change the file_type in the trace this issue is solved.